src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/matcher/AttributeMatcher.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.ccd.domain.service.accessprofile.filter.matcher;
+
+import java.util.Optional;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+public interface AttributeMatcher {
+
+    boolean matchAttribute(RoleAssignmentFilteringResult result, CaseDetails caseDetails);
+
+    default boolean isValuesMatching(Optional<String> roleAssignmentValue,
+                                     String caseDataValue) {
+        if (roleAssignmentValue == null) {
+            return true;
+        }
+        return roleAssignmentValue.get() == caseDataValue
```

Comment:
how come `roleAssignmentValue.get() == caseDataValue`? isn't the equals enough?

Reply:
roleAssignmentValue.get() == caseDataValue changed to roleAssignmentValue.get() == null